### PR TITLE
Wrap todo descriptions at word boundaries in sidebar

### PR DESCRIPTION
**Pull request by rumpl**
Replace truncation with word-aware wrapping so that todo items in the sidebar display their full description regardless of sidebar width.

Long descriptions wrap onto subsequent lines, indented to align with the text after the icon prefix, and line breaks prefer word boundaries over splitting mid-word.

### Changes

- **`pkg/tui/components/toolcommon/truncate.go`** — New `WrapLinesWords` function that wraps at word boundaries, falling back to rune-level splitting for words exceeding the available width.
- **`pkg/tui/components/tool/todotool/sidebar.go`** — Use `WrapLinesWords` instead of `TruncateText` so todo descriptions wrap fully with proper indentation on continuation lines.
- **`pkg/tui/components/toolcommon/common_test.go`** — Tests for `WrapLinesWords` covering normal wrapping, long words, multi-line input, edge cases, and unicode/CJK.

Screenshot 2026-02-08 at 23 08 14

### CODE REVIEW

File: pkg/tui/components/tool/todotool/sidebar.go [hunk 1]
```diff
@@ -58,12 +58,23 @@ func (c *SidebarComponent) Render() string {
 func (c *SidebarComponent) renderTodoLine(todo builtin.Todo) string {
 	icon, style := renderTodoIcon(todo.Status)
 
-	// Compute prefix width dynamically (icon + space separator)
 	prefix := icon + " "
-	maxDescWidth := c.width - lipgloss.Width(prefix)
-	description := toolcommon.TruncateText(todo.Description, maxDescWidth)
+	prefixWidth := lipgloss.Width(prefix)
+	maxDescWidth := c.width - prefixWidth
+
```
[the remark this answers is not in the document]
Comment (rumpl) [reply pkg/tui/components/tool/todotool/sidebar.go:64]: Fixed

**Comment by rumpl** [comment]
Adding separation will make it even more vertical, idk

**Comment by krissetto** [comment]
true, but maybe even todos should have a title and content if they are that long though, these in the example are more like full blown tasks than simple todos.. but anyway stuff for separate PRs
